prime-router/src/main/kotlin/cli/tests/SftpcheckTest.kt

```diff
@@ -0,0 +1,159 @@
+        val ignoreReceiverNamePath = environment.formUrl(ignoreReceiverNamesURI).toString()
+        val ignoreReceiversNameList = listReceiverNames(ignoreReceiverNamePath, accessToken)
+
+        /**
+         * If the Ignore receiver list is empty, prompt error message and skip the test.
+         */
+        if (ignoreReceiversNameList.isEmpty()) {
+            return bad(sftpcheckMessage + "FAILED: Ignore receiver list is empty.")
+        }
+
+        /**
+         * Start check the connection for each organization
+         */
+        ignoreReceiversNameList.forEach { receiver ->
+            /**
+             * Obtain the URL/path endpoint per environment (localhost or staging)
+             */
+            val path = environment.formUrl(sftpcheckUri + receiver).toString()
+
+            /**
+             * Check the organization ignore receiver connections
+             */
+            echo("SFTPCHECK Organizatin: $receiver...")
+
+            val (_, response, result) = sftpReceiverIgnoreOrganizationCheck(path, accessToken)
+            val (_, error) = result
+            if (response.statusCode == HttpStatus.SC_OK) {
+                good(
+                    sftpcheckMessage + "PASSED with response code: " +
+                        " ${response.statusCode} "
+                )
+            } else {
+                sftpcheckTestResult = bad(
+                    sftpcheckMessage + "FAILED with error code: : " +
+                        "${response.statusCode} - ${error?.response?.responseMessage}..."
+                )
+            }
+        }
+
+        return when (sftpcheckTestResult) {
+            true -> good("Test passed: SFTPCHECK ")
+            false -> bad(sftpcheckMessage + "Failed")
+        }
+    }
+
+    /**
+     * listReceiverNames - it extracts all receiver ignore organization from the database setting table
+     *  using setting API.
+     *
+     * @returns:
+     *  receiverNames - contains all receiver ignore organizations
+     *  emptyList - if there is any problem with getting ignore organization.
+     */
+    fun listReceiverNames(
+        path: String,
+        accessToken: String
+    ): List<String> {
+        val (_, _, result) = Fuel
+            .get(path)
+            .authentication()
+            .bearer(accessToken)
+            .header(Headers.CONTENT_TYPE to jsonMimeType)
+            .responseJson()
+        return when (result) {
+            is Result.Failure -> emptyList()
+            is Result.Success -> {
+                val receiverJsonArray = result.value.array()
+                (0 until receiverJsonArray.length())
+                    .map { receiverJsonArray.getJSONObject(it) }
+                    .filter {
+                        (
+                            !it.isNull("transport") &&
+                                !it.getJSONObject("transport").isNull("host") &&
+                                it.getJSONObject("transport").getString("host").equals("sftp")
```

Comment:
Could this be consolidated something like:

it.getJSONObject("transport")?.getString("host") == "sftp"